ArduCopter/mode_auto.cpp

```diff
@@ -833,6 +852,8 @@ void ModeAuto::wp_run()
     if (auto_yaw.mode() == AUTO_YAW_HOLD) {
         // roll & pitch from waypoint controller, yaw rate from pilot
         attitude_control->input_thrust_vector_rate_heading(wp_nav->get_thrust_vector(), target_yaw_rate);
+    } else if (auto_yaw.mode() == AUTO_YAW_RATE) {
```

Comment:
By making it a SWITCH statement, I think it will be clear that it is a difference in mode processing.